@jherland failure on aarch64-linux

```
bench/bench.cpp:32:10: error: #error no rdtsc implementation
   32 |         #error no rdtsc implementation
      |          ^~~~~
bench/bench.cpp: In function 'int main()':
bench/bench.cpp:141:38: error: 'rdtsc' was not declared in this scope
  141 |                     uint64_t start = rdtsc();
      |                                      ^~~~~
error: builder for '/nix/store/9hxpk9nv2hfgyc96q51gqmr7krkk8jgq-pdqsort-bench-0-unstable-2021-03-14.drv' failed with exit code 1;
       last 17 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/7q1qgmyqg2kwqz5lprpbj5r88rzw0pyx-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > no Makefile or custom buildPhase, doing nothing
       > Running phase: checkPhase
       > bench/bench.cpp:32:10: error: #error no rdtsc implementation
       >    32 |         #error no rdtsc implementation
       >       |          ^~~~~
       > bench/bench.cpp: In function 'int main()':
       > bench/bench.cpp:141:38: error: 'rdtsc' was not declared in this scope
       >   141 |                     uint64_t start = rdtsc();
       >       |                                      ^~~~~
       For full logs, run:
         nix log /nix/store/9hxpk9nv2hfgyc96q51gqmr7krkk8jgq-pdqsort-bench-0-unstable-2021-03-14.drv
```

--- a/pkgs/by-name/pd/pdqsort/package.nix
+++ b/pkgs/by-name/pd/pdqsort/package.nix
@@ -1,0 +1,49 @@
+{
+  lib,
+  stdenvNoCC,
+  stdenv, # for tests
+  fetchFromGitHub,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "pdqsort";
+  version = "0-unstable-2021-03-14";
+
+  src = fetchFromGitHub {
+    owner = "orlp";
+    repo = "pdqsort";
+    rev = "b1ef26a55cdb60d236a5cb199c4234c704f46726";
+    hash = "sha256-xn3Jjn/jxJBckpg1Tx3HHVAWYPVTFMiDFiYgB2WX7Sc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/include
+    cp -r *.h $out/include/
+
+    runHook postInstall
+  '';
+
+  # The benchmark takes too long to run as a regular checkPhase here.
+  passthru.tests.bench = stdenv.mkDerivation {
+    pname = "pdqsort-bench";
+
+    inherit (finalAttrs) version src;
+
+    doCheck = true;
+    checkPhase = ''
+      c++ bench/bench.cpp -o bench/bench
+      ./bench/bench > $out
+    '';
+
+    meta.platforms = lib.platforms.x86_64;
+  };
+
+  meta = {
+    description = "Novel sorting algorithm that combines the fast average case of randomized quicksort with the fast worst case of heapsort";
+    homepage = "https://github.com/orlp/pdqsort";
+    license = lib.licenses.zlib;
+    maintainers = with lib.maintainers; [ jherland ];
+  };
+})

